SConstruct

```diff
@@ -690,7 +553,212 @@ config_options = [
+config = Configuration()
+
+if "help" in COMMAND_LINE_TARGETS:
+    AddOption(
+        "--list-options", dest="list",
+        action="store_true", help="List available options")
+    AddOption(
+        "--restructured-text", dest="rest",
+        action="store_true", help="Format defaults as ReST")
+    AddOption(
+        "--option", dest="option", nargs=1, type="string",
+        action="store", help="Output help for specific option")
+    AddOption(
+        "--defaults", dest="defaults",
+        action="store_true", help="All defaults (CLI only)")
+    AddOption(
+        "--dev", dest="dev",
+        action="store_true", help="Append -dev (ReST only)")
+    AddOption(
+        "--output", dest="output", nargs=1, type="string",
+        action="store", help="Output file (ReST only)")
+
+    list = GetOption("list")
+    rest = GetOption("rest")
+    defaults = GetOption("defaults") is not None
+
+    if defaults or rest or list:
+
+        config.add(windows_options)
+        config.add(config_options)
+        option = GetOption("option")
+
+        if list:
+            logger.info("Avalable configuration options:")
+            logger.info(config.list(), print_level=False)
+            sys.exit(0)
+
+        if defaults:
+            try:
+                logger.info(config.help(option), print_level=False)
+                sys.exit(0)
+            except KeyError as err:
+                message = "Run 'scons help --list-options' to see available options."
+                logger.error(f"{err}.\n{message}")
+                sys.exit(1)
+
+        dev = GetOption("dev") is not None
+        try:
+            message = config.to_rest(option, dev=dev)
+        except KeyError as err:
+            message = "Run 'scons help --list-options' to see available options."
+            logger.error(f"{err}.\n{message}")
+            sys.exit(1)
+
+        output = GetOption("output")
+        if output:
+            output_file = Path(output).with_suffix(".rst")
+            with open(output_file, "w+") as fid:
+                fid.write(message)
+
+            logger.info(f"Done writing output options to '{output_file}'.",
+                        print_level=False)
+
+        else:
+            logger.info(message, print_level=False)
+
+        sys.exit(0)
+
+# **************************************
+# *** Read user-configurable options ***
+# **************************************
+
+opts = Variables("cantera.conf")
+
+extraEnvArgs = {}
+
+if os.name == "nt":
+    config.add(windows_options)
+    config.add(config_options)
+
+    config["prefix"].default = pjoin(os.environ["ProgramFiles"], "Cantera")
+    config.select("Windows")
+
+    # On Windows, target the same architecture as the current copy of Python,
+    # unless the user specified another option.
+    if "64 bit" not in sys.version:
+        config["target_arch"].default = "x86"
+
+    opts.AddVariables(*config.to_scons(["msvc_version", "target_arch"]))
```

Comment:
Could this be a tuple? Although it probably doesn't make a difference, tuples are slightly faster, and semantically indicate the argument won't be change.
```suggestion
    opts.AddVariables(*config.to_scons(("msvc_version", "target_arch")))
```